spec/ParseLiveQuery.spec.js

```diff
@@ -969,8 +969,10 @@ describe('ParseLiveQuery', function () {
     await user.save();
     user.set('yolo', 'bar');
     await user.save();
+    user.set('foo', 'bar');
+    await user.save();
     await user.destroy();
-    await new Promise(resolve => process.nextTick(resolve));
+    await new Promise(resolve => setTimeout(resolve, 500));
```

Comment:
Do we need 500ms timeout? That's quite a long time.

Reply:
It might not, but it's a CI test and I wanted to be on the save side. Even if it's reduced, it has close to zero impact on the total CI execution time.

Reply:
Do I read this wrong or does this PR contain 3 x 500ms timeouts? If so, I wouldn't agree that 1.5s is "close to zero". Could you try to reduce this to 10ms?

Reply:
Yes, it's 1.5sec on a CI task running about 500sec. In percentage that's close to zero. If the task fails because the timeouts being too short, the overall penalty of re-running the workflow is much higher.

Is there a way for me to start the Workflows myself? Than I'm happy to experiment a bit with lower values.. (will test it locally but like to start the workflow immediately after committing so I can adjust when needed)

Reply:
Please reduce to 10ms and I will run the CI a couple of times. If it passes, we'll take it.

Reply:
Just looked again at the code, ParseLiveQuery,spec.js has a 1000msec timeout after _every_ test:

```
  afterEach(async function (done) {
    const client = await Parse.CoreManager.getLiveQueryController().getDefaultLiveQueryClient();
    client.close();
    // Wait for live query client to disconnect
    setTimeout(() => {
      done();
    }, 1000);
  });
```

Sorry but I honestly don't want to spend more time on this. If you like CI to be time/performance optimised then I suggest to create a separate issue for that. For example, the duplicate "should strip out session token in LiveQuery" test can probably be removed from ParseUser.spec.js? The 'afterEach' timeout in ParseLiveQuery.spec.js might be set to 500 or lower instead of 1000msec? I'm sure there are a bunch of other optimisations possible to shave of some CI time and cpu cycles.. 

I was more than happy to contribute but really don't feel like going back and forth on experimenting with timeouts as part of a PR discussion ("500 to much, 10 to short, 100 maybe, lets try 150, what about 120..."). Hope you understand..

Reply:
I understand this can be tedious, no worries I will take care of this. 10ms should be more than enough. Compute time on GitHub is not unlimited and free of charge, so we have to be diligent about it. If you want to optimize other tests, we would be happy if you want to open a separate PR.